### ISSUE TITLE
Playlist header columns not redrawn after widget resize

### ISSUE THREAD

**Issue by Lithopsian** [Alexey-Yakovenko/deadbeef#1308]
To reproduce:
1. Size the playlist widget to include a horizontal scrollbar.
2. Scroll the playlist all the way to the left (ie. scrollbar slider to the right).
3. Resize the playlist widget until the horizontal scrollbar disappears (and quite a bit more makes the bug easier to spot).
The playlist column headers are left in a state that is not consistent with the columns themselves.  They are still partially scrolled even though the columns themselves are not.
4. Resize back until the horizontal scrollbar reappears.
The playlist columns and headers are now consistent and set to their starting positions, but the scrollbar indicates that they are still scrolled.  It helps to resize quickly.

**Comment by Lithopsian** [comment]
There is also a drawing artefact on the horizontal scrollbar after some resizes.  I can reproduce it fairly consistently:
1. Make the playlist quite narrow and scroll all the way to the right.
2. Quickly resize the playlist so that the horizontal scrollbar disappears.
3. Gradually resize it narrower until the scrollbar reappears.  The scrollbar slider is drawn over the right hand button, although the button arrow is still visible.

This only seems to happen in GTK2 so could be considered a GTK bug.  It seems to happen because the position value passed to the adjustment constructor is not clamped and can exceed the largest legal value.  I can fix that, and then the bug should go away.  Values set after the adjustment has been added to the scrollbar appear to be clamped correctly in versions I tested.
